contracts/mainnet/connectors/swap-aggregator/main.sol

```diff
@@ -0,0 +1,53 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import "./helpers.sol";
+import "./events.sol";
+
+contract SwapAggregatorConnector is Helpers, Events {
+    function swap(
+        address buyAddr,
+        address sellAddr,
+        uint256 sellAmt,
+        uint256 unitAmt,
+        bytes calldata callData,
+        address routerAddress,
+        address allowanceHolderAddress,
```

Comment:
@SamarendraGouda We won't pass any address, instead, we will manage a mapping on this connector itself.

string connectorName => struct(router, allowance). This can only be updated by the chief of connector or master address